src/neomaril_codex/training.py

```diff
@@ -248,10 +248,10 @@ def _set_params(self):
             params = {**params, **hyperparameters}
 
         if len(self.y_train.value_counts()) < 10:
-            target_proportion = (
-                self.y_train.value_counts() / len(self.y_train)
-            ).to_dict()
-
+            
+            target_proportion = self.y_train.value_counts() / len(self.y_train)
+            target_proportion = target_proportion.to_dict()
+            target_proportion = { str(k[0]) : v for k, v in target_proportion.items() }
```

Comment:
Então, nesse caso fica uma tupla pq classificação nem sempre é binária, e nem sempre as classes são 0 e 1. Se vai dar problema com a tupla melhor fazer um `str(k)` do que pegar só a primeira classe

Reply:
Acontece que a estrutura era `{(0,): 0.3, (1,) : 0.7}`, a lógica não está impedindo que tenha mais valores, só está pegando o primeiro valor da tupla e transformando em chave: `{0: 0.3, 1 : 0.7}`

Reply:
[Aqui](https://www.kaggle.com/competitions/lish-moa/data?select=train_targets_nonscored.csv) um exemplo de multitarget

Reply:
Entendi! Estava pensando que estava tratando dessa estrutura aqui: `{ (0,): 0.2, (1,) : 0.5, (2,) : 0.3}`

Reply:
Acho que seria interessante mudar a estrutura de um `{ (1,) : 0.5 }` para `{ 'target' : (1,), 'proportion' : 0.5 } `

Reply:
Solved in the last commit!